### PR TITLE
fix IAM joining failing in some cases when a web identity was used as an AWS credential

### PR DESCRIPTION
**Pull request by capnspacehook**
When the Teleport service attempting to join a cluster using an IAM token was running in FIPS mode in an AWS region without a FIPS STS endpoint, and a web identity token was used as the AWS credential, the local AWS region would be used to build the STS request. We had logic to detect this and fallback to the `us-east-1` region instead, but that was done by passing an additional config to `sts.New`. The config that the AWS `session.Session` contains takes priority over all others. The `us-east-1` region was initially used to build the STS request, but when calling `Request.Sign` on the STS request the URL was changed to use the region configured in the `session.Session`. 

Instead, rebuild the `session.Session` and STS client `sts.STS` when the region needs to be changed.

Fixes https://github.com/gravitational/customer-sensitive-requests/issues/308.

changelog: fix IAM joining failing in some cases when a web identity was used as an AWS credential

### CODE REVIEW

**Review by rosstimothy — COMMENTED** [review]
Would migrating the sts client to aws-sdk-go-v2 allow this problem to be solved in a different manner? Would this even be a problem with the aws-sdk-go-v2 sts client?

File: lib/auth/join/iam/iam.go
Comment (nklaassen) [inline lib/auth/join/iam/iam.go:1]: i'm not sure if this is unit testable, probably tricky, maybe we should add this case to the test plan?

**Comment by nklaassen** [comment]
I basically rewrote this function to use aws-sdk-go-v2 in https://github.com/gravitational/teleport/pull/47044. The new implementation doesn't have this bug (and I wrote a unit test for it). I believe this PR can be closed.

**Comment by capnspacehook** [comment]
Can confirm upgrading to `aws-sdk-go-v2` fixed the issue, this PR is not needed

**Pull request closed without merging**